chart-values-json-schema-validation.py

```diff
@@ -0,0 +1,180 @@
+    """
+    Download json schema for chart version.
+    """
+    version_url = f"{SCHEMA_BASE_URL}/version.json"
+    version_json = requests.get(version_url, timeout=10, verify=False).json()
+    return_version = next(
+        (
+            item["JSONSchemaVersion"]
+            for item in version_json["ranges"]
+            if item["end"] >= version >= item["start"]
+        ),
+        None,
+    )
+    if return_version:
+        schema_url = f"{SCHEMA_BASE_URL}/v{return_version}/schema-platform-managed-chart-strict.json"
+        schema_json = requests.get(schema_url, timeout=10, verify=False).json()
+        return schema_json
+    return None
+
+
+def delete_error_files(directory="."):
+    """
+    Delete error files.
+    """
+    for filename in os.listdir(directory):
+        if filename.startswith("error-merged-values-") and filename.endswith(".yaml"):
+            file_path = os.path.join(directory, filename)
+            try:
+                os.remove(file_path)
+            except Exception as err:
+                print(f"Error deleting {file_path}. Reason: {err}")
+                continue
+
+
+def deep_merge(source, destination):
+    """
+    Deep merge dictionaries.
+    """
+    for key, value in source.items():
+        if isinstance(value, dict):
+            # Get node or create one
+            node = destination.setdefault(key, {})
+            deep_merge(value, node)
+        else:
+            destination[key] = value
+
+    return destination
+
+
+def merge_yaml_files(svc_path, instance_file):
+    """
+    Merge values.yaml, values-env.yaml and values-env-instance.yaml files.
+    """
+    yaml = YAML()
+    merged_data = {}
+
+    # Base file
+    with open(os.path.join(svc_path, "values.yaml"), "r", encoding="utf8") as base_file:
+        merged_data.update(yaml.load(base_file))
+
+    # Environment file
+    env = "dev" if "dev" in instance_file else "prod"
+    env_file_path = os.path.join(svc_path, f"values-{env}.yaml")
+    if os.path.exists(env_file_path):
+        with open(env_file_path, "r", encoding="utf8") as env_file:
+            env_data = yaml.load(env_file)
+            deep_merge(env_data, merged_data)
+
+    # Instance file
+    with open(
+        os.path.join(svc_path, instance_file), "r", encoding="utf8"
+    ) as instance_f:
+        instance_data = yaml.load(instance_f)
+        deep_merge(instance_data, merged_data)
+
+    return merged_data
+
+
+def validate_json_schema(merged_values, schema_data):
+    """
+    Validate merged values against json schema.
+    """
+    validate(instance=merged_values, schema=schema_data)
+
+
+def main():
+    """
+    Main function.
+    """
+    if not os.path.exists(GITOPS_DIR):
+        print(f"{GITOPS_DIR} directory is missing, exiting...")
+        sys.exit(0)
+
+    for app in os.listdir(GITOPS_DIR):
+        app_path = os.path.join(GITOPS_DIR, app)
+        if os.path.isdir(app_path):
+            for svc in os.listdir(app_path):
```

Comment:
I think you can directly use glob to find the final files.

Reply:
changed to use glob to search files